pkg/apis/v1alpha1/dnspolicy_types.go

```diff
@@ -34,20 +34,18 @@ type DNSPolicySpec struct {
 	// +optional
 	HealthCheck *HealthCheckSpec `json:"healthCheck,omitempty"`
 
-	// +kubebuilder:validation:Required
-	// +required
+	// +optional
 	LoadBalancing *LoadBalancingSpec `json:"loadBalancing"`
 }
 
 type LoadBalancingSpec struct {
-	// +required
+	// +optional
 	Weighted *LoadBalancingWeighted `json:"weighted,omitempty"`
 	// +optional
 	Geo *LoadBalancingGeo `json:"geo,omitempty"`
 }
 
 // +kubebuilder:validation:Minimum=0
-// +kubebuilder:validation:Maximum=255
 type Weight int
```

Comment:
Removed validation on upper value for weights. Since we will now accept anything for geo, seemed to make sense that we don't limit what you can set for weight either. Providers have different values they will accept:

- Route53: 0 - 255
- Google: 0 - 10000
- Azure:  1- 1000

Like Geo, adding an invalid value for a provider will result in a reconcile error. Error returned from the provider is in the message of the condition.